include/merlin_hashtable.cuh

```diff
@@ -50,10 +50,24 @@ namespace merlin {
  * @note Eviction occurs automatically when a bucket is full. The keys with the
  * minimum `score` value are evicted first.
  *
+ * @note For `kEpochLru` and `kEpochLfu`, the high 32bits would be set to
+ * `global_epoch` while the low 32 bits is `timestamp` or `frequency`.
+ *
+ * @note on `kLFU`, when the scores reaches to the max of `uint64_t`, it will
+ * not increase any more.
+ *
+ * @note on `kLkEpochLfu`, when the lower 32bit stand for frequency reaches to
+ * the max of `uint32_t`, it will not increase any more.
+ *
  */
-enum class EvictStrategy {
-  kLru = 0,        ///< LRU mode.
-  kCustomized = 1  ///< Customized mode.
+struct EvictStrategy {
```

Comment:
Since EvictStategy is a template param for HashTable, is it very need to use it as a constexpr static int, instead of EvictStategy::kEpochLru? It not, maybe use a macro to make the template param meaningful.

Reply:
Using the structure mainly serves for conveniently exposing the `API` to and being easily referred by the end-users via `EvictStategy`, and I believe the macro has no name scope and can dirty the user namespace.

Reply:
But each enum member is itself a constexpr.I vote for enum class. In addition, how about setting kCustomized to 0, the next time you add a new stragy, you can add it directly, and it looks more natural.

Reply:
I have re-designed it, please refer to the latest code.